web/server/codechecker_server/api/product_server.py

```diff
@@ -90,26 +89,23 @@ def __get_product(self, session, product):
             self.__server.add_product(product)
             server_product = self.__server.get_product(product.endpoint)
 
-        server_product.connect()
         num_of_runs = 0
         runs_in_progress = set()
         latest_store_to_product = ""
-        if server_product.db_status == shared.ttypes.DBStatus.OK:
-            with DBSession(server_product.session_factory) as run_db_session:
-                run_locks = run_db_session.query(RunLock.name) \
-                    .filter(RunLock.locked_at.isnot(None)) \
-                    .all()
-
-                runs_in_progress = set([run_lock[0] for run_lock in run_locks])
-
-                num_of_runs = run_db_session.query(Run).count()
-                if num_of_runs:
-                    last_updated_run = run_db_session.query(Run) \
-                        .order_by(Run.date.desc()) \
-                        .limit(1) \
-                        .one_or_none()
-
-                    latest_store_to_product = last_updated_run.date
+
+        # Try to get product details. It may throw an error in case of
+        # connection error, so we should try to connect to the database
```

Comment:
Please add a similar comment to the `get_details` function, that it might throw different exceptions depending on the used database adapter.